eclipse/src/saros/ui/eventhandler/ConnectingFailureHandler.java

```diff
@@ -19,66 +18,56 @@
  */
 public class ConnectingFailureHandler implements IConnectingFailureCallback {
 
-  private static final Logger LOG = Logger.getLogger(ConnectingFailureHandler.class);
+  private static final Logger log = Logger.getLogger(ConnectingFailureHandler.class);
 
   private final ConnectionHandler connectionHandler;
-  private final XMPPAccountStore accountStore;
 
   private boolean isHandling;
 
-  public ConnectingFailureHandler(
-      final ConnectionHandler connectionHandler, final XMPPAccountStore accountStore) {
+  public ConnectingFailureHandler(final ConnectionHandler connectionHandler) {
     this.connectionHandler = connectionHandler;
     this.connectionHandler.setCallback(this);
-    this.accountStore = accountStore;
   }
 
   @Override
   public void connectingFailed(final Exception exception) {
-    SWTUtils.runSafeSWTAsync(LOG, () -> handleConnectionFailed(exception));
+    SWTUtils.runSafeSWTAsync(log, () -> handleConnectionFailed(exception));
   }
 
   private void handleConnectionFailed(Exception exception) {
 
-    // account store is empty
-    if (exception == null) {
-      // Wizard will perform a connection attempt if it is finished
-      SWTUtils.runSafeSWTAsync(LOG, () -> WizardUtils.openSarosConfigurationWizard());
-      return;
-    }
+    try {
 
-    // avoid mass dialog popups
-    if (isHandling) return;
+      // avoid mass dialog popups
+      if (isHandling) return;
 
-    try {
       isHandling = true;
 
       if (!(exception instanceof XMPPException)) {
-
-        DialogUtils.popUpFailureMessage(
+        MessageDialog.openError(
+            SWTUtils.getShell(),
             "Connecting Error",
             MessageFormat.format(
-                "Could not connect to XMPP server. Unexpected error: {0}", exception.getMessage()),
-            false);
-
+                "Could not connect to XMPP server. Unexpected error: {0}", exception.getMessage()));
         return;
       }
 
-      if (DialogUtils.popUpYesNoQuestion(
-          "Connecting Error",
-          generateHumanReadableErrorMessage((XMPPException) exception),
-          false)) {
+      final XMPPAccount accountUsedForConnecting =
+          XMPPConnectionSupport.getInstance().getCurrentXMPPAccount();
 
-        /* FIXME the active/default account might not always be the account that is currently used for connecting */
-        final XMPPAccount accountUsedDuringConnection = accountStore.getDefaultAccount();
+      // should not happen
+      if (accountUsedForConnecting == null) return;
```

Comment:
Log a warning to show that an unexpected state occurred in order to allow us to noticed/investigated this behavior?